File: sample_code/vulnerable.java
import java.sql.*;

public class VulnerableJava {
    
    // SQL Injection vulnerability
    public static ResultSet getUserData(String username) throws SQLException {
        String query = "SELECT * FROM users WHERE username = '" + username + "'";
        Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/db");
        Statement stmt = conn.createStatement();
        return stmt.executeQuery(query);
    }
    
    // Hardcoded credentials
    private static final String DB_PASSWORD = "admin123";
    private static final String API_KEY = "sk-1234567890abcdef";
    
    // Weak random number generation
    public static String generateToken() {
        java.util.Random rand = new java.util.Random();
        return String.valueOf(rand.nextInt(100000));
    }
    
    // Command injection
    public static void executeCommand(String cmd) throws Exception {
        Runtime.getRuntime().exec(cmd);
    }
    
    // Insecure deserialization
    public static Object deserialize(byte[] data) throws Exception {
        java.io.ObjectInputStream ois = new java.io.ObjectInputStream(
            new java.io.ByteArrayInputStream(data)
        );
        return ois.readObject();
    }
}

File: sample_code/vulnerable.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Buffer overflow vulnerability
void process_input(char *input) {
    char buffer[10];
    strcpy(buffer, input);  // Unsafe - no bounds checking
    printf("Buffer: %s\n", buffer);
}

// Command injection vulnerability
void execute_command(char *cmd) {
    system(cmd);  // Unsafe - user input passed directly
}

// Use of uninitialized variable
int unsafe_math() {
    int result;
    result = result + 5;  // result not initialized
    return result;
}

// Hardcoded credentials
const char *PASSWORD = "admin123";
const char *API_KEY = "sk-1234567890abcdef";